python/nav/web/netmap/graph.py

```diff
@@ -134,13 +159,30 @@ def get_traffic_interfaces(edges, interfaces):
     return storage.values()
 
 
-def get_layer2_traffic():
+def get_layer2_traffic(locationOrRoomId):
     """Fetches traffic data for layer 2"""
+    # Cache model: Index traffic by location
+    cache_key = _cache_key("traffic", "locationId", "layer 2")
```

Comment:
Great suggestion, and I agree that this should not be a concern of the functions themselves. I implemented a couple of caching decorators and moved them into a separate file (`cache.py`).